internal/command/postgres/update.go

```diff
@@ -150,43 +191,34 @@ func runUpdate(ctx context.Context) error {
 
 	// Update replicas
 	if len(replicas) > 0 {
-		fmt.Fprintf(io.Out, "Updating replicas\n")
+		fmt.Fprintf(io.Out, "Updating replica(s)\n")
 
 		for _, replica := range replicas {
-			ref := updateList[replica.ID]
-
-			if ref == nil {
-				fmt.Fprintf(io.Out, "  Machine %s is already running the latest image\n", replica.ID)
-				continue
-			}
-
-			image := fmt.Sprintf("%s:%s", ref.Repository, ref.Tag)
+			image := fmt.Sprintf("%s:%s", latest.Repository, latest.Tag)
 
-			fmt.Fprintf(io.Out, "  Updating machine %s with image %s %s\n", replica.ID, image, ref.Version)
+			fmt.Fprintf(io.Out, "  Updating machine %s with image %s %s\n", replica.ID, image, latest.Version)
 			if err := updateMachine(ctx, app, replica, image); err != nil {
 				return fmt.Errorf("can't update %s: %w", replica.ID, err)
 			}
 		}
 	}
 
 	// Update leader
-	ref := updateList[leader.ID]
-	if ref != nil {
-		// Don't perform failover if the cluster is only running a
-		// single node.
+	if leader != nil {
+		// Only perform failover if there's potentially eligible replicas
 		if len(machines) > 1 {
 			pgclient := flypg.New(app.Name, dialer)
 
-			fmt.Fprintf(io.Out, "Performing a failover\n")
+			fmt.Fprintf(io.Out, "Performing failover\n")
 			if err := pgclient.Failover(ctx); err != nil {
 				return fmt.Errorf("failed to trigger failover %w", err)
 			}
 		}
 
 		fmt.Fprintf(io.Out, "Updating leader\n")
```

Comment:
nitpick: the message is miss leading (pun-intended) in case of a failover  🔍

Reply:
heh, the weird part is that the failover isn't happening for single node postgres. Not a big deal